stf/model/trainer/lightgbm/lightgbm.py

```diff
@@ -0,0 +1,452 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+import pytz
+import lightgbm as lgb
+from ktpbase.log import logging
+
+from stf.model import metrics
+from stf.model.general import (
+    pre_process_data, remove_features_not_in_set, split_data_train_validation_test
+)
+from stf.model.trainer.trainer import AbstractModelTrainer
+
+# Available trainings period durations for optimization
+# After preprocessing, the data consists of 75 days (of the original 90 days).
+# To prevent overfitting on a very short period of time, the test and validation
+# sets are always a constant fraction of the total data period (75 days).
+# We take an optimized fraction of the remaining data as our train-data.
+# Given the current fractions for test and validation, the longest train-block
+# is 60 days.
+
+
+class LGBModelTrainer(AbstractModelTrainer):
+    def __init__(self, pj):
+        super().__init__(pj)
+        self.logger = logging.get_logger(self.__class__.__name__)
+        # See LGBoost Parameters
+        # https://lightgbm.readthedocs.io/en/latest/Python-Intro.html#setting-parameters
+        # look at the parameters for Tree booster
+        self.hyper_parameters = {
+            "max_depth": 8
+        }
+        # self.hyper_parameters = {
+        #     # general model specific hyper paramater (NOT optimized)
+        #     # "silent": 1,
+        #     "disable_default_eval_metric": 1,
+        #     # model specific hyper parameters (optimized)
+        #     "subsample": 0.9,
+        #     "min_child_weight": 4,
+        #     "max_depth": 8,
+        #     "gamma": 0.5,
+        #     "colsample_bytree": 0.85,
+        #     "eta": 0.1,
+        #     # model specific hyper parameters (NOT optimized)
+        #     "objective": "reg:squarederror",
+        #     # generic hyper parameters
+        #     "featureset_name": "G",
+        #     "training_period_days": 90,
+        # }
+        # private key lists for filtering
+        self._lgb_hyper_parameter_keys = [
+            # "silent",
+            # "disable_default_eval_metric",
+            # "subsample",
```

Comment:
Are these comments nescessary?

Reply:
yes. added todo to fill them with correct values